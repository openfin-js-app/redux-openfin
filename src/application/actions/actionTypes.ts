import makeType from '../../utils/makeType';

export const NEW_APPLICATION = makeType('NEW_APPLICATION');
export const NEW_APPLICATION_RES = makeType('NEW_APPLICATION_RES');

export const RESTART = makeType('APPLICATION_RESTART');
export const RESTART_RES = makeType('APPLICATION_RESTART_RES');

export const CLOSE = makeType('APPLICATION_CLOSE');
export const CLOSE_RES = makeType('APPLICATION_CLOSE_RES');
