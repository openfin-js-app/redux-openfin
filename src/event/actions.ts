import {EventConstant, appEvents, windowEvents} from './constants';

export type ActionDict = {
    [name:string]:EventConstant,
}

const appEventDictByName:ActionDict = {};
const appEventDictByType:ActionDict = {};
const windowEventDictByName:ActionDict = {};
const windowEventDictByType:ActionDict = {};

appEvents.forEach(oneEvent => {
    appEventDictByName[oneEvent.name] = oneEvent;
    appEventDictByType[oneEvent.type] = oneEvent;
});

windowEvents.forEach(oneEvent => {
    windowEventDictByName[oneEvent.name] = oneEvent;
    windowEventDictByType[oneEvent.type] = oneEvent;
});

export default {
    appEventDictByName,
    appEventDictByType,
    windowEventDictByName,
    windowEventDictByType,
};