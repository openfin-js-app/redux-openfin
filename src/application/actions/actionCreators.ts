import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const startApplication:ActionCreator<types.StartApplicationPayload> = createFSA<types.StartApplicationPayload>(
    actionTypes.START_APPLICATION,
    (options:types.StartApplicationPayload) => <any> options
);

export const getCurrent:ActionCreator<types.GetCurrentPayload> = createFSA<types.GetCurrentPayload>(
    actionTypes.APPLICATION_GET_CURRENT,
    (options:types.GetCurrentPayload) => <any> options
);

export const getWindow:ActionCreator<types.GetWindowPayload> = createFSA<types.GetWindowPayload>(
    actionTypes.APPLICATION_GET_WINDOW,
    (options:types.GetWindowPayload) => <any> options
);

export const wrap:ActionCreator<types.WrapPayload> = createFSA<types.WrapPayload>(
    actionTypes.APPLICATION_WRAP,
    (options:types.WrapPayload) => <any> options
);

export const addEventListener:ActionCreator<types.AddListenerPayload> = createFSA<types.AddListenerPayload>(
    actionTypes.APPLICATION_ADD_LISTENER,
    (options:types.AddListenerPayload) => <any> options
);

export const quit:ActionCreator<types.QuitPayload> = createFSA<types.QuitPayload>(
    actionTypes.APPLICATION_QUIT,
    (options:types.QuitPayload) => <any> options
);

export const getChildWindows:ActionCreator<types.GetChildWindowsPayload> = createFSA<types.GetChildWindowsPayload>(
    actionTypes.APPLICATION_GET_CHILD_WINDOWS,
    (options:types.GetChildWindowsPayload) => <any> options
);

export const getInfo:ActionCreator<types.GetInfoPayload> = createFSA<types.GetInfoPayload>(
    actionTypes.APPLICATION_GET_INFO,
    (options:types.GetInfoPayload) => <any> options
);

export const getShortcuts:ActionCreator<types.GetShortcutsPayload> = createFSA<types.GetShortcutsPayload>(
    actionTypes.APPLICATION_GET_SHORTCUTS,
    (options:types.GetShortcutsPayload) => <any> options
);

export const getTrayIconInfo:ActionCreator<types.GetTrayIconInfoPayload> = createFSA<types.GetTrayIconInfoPayload>(
    actionTypes.APPLICATION_GET_TRAY_ICON_INFO,
    (options:types.GetTrayIconInfoPayload) => <any> options
);

export const getZoomLevel:ActionCreator<types.GetZoomLevelPayload> = createFSA<types.GetZoomLevelPayload>(
    actionTypes.APPLICATION_GET_ZOOM_LEVEL,
    (options:types.GetZoomLevelPayload) => <any> options
);

export const isRunning:ActionCreator<types.IsRunningPayload> = createFSA<types.IsRunningPayload>(
    actionTypes.APPLICATION_IS_RUNNING,
    (options:types.IsRunningPayload) => <any> options
);

export const removeListener:ActionCreator<types.RemoveListenerPayload> = createFSA<types.RemoveListenerPayload>(
    actionTypes.APPLICATION_REMOVE_LISTENER,
    (options:types.RemoveListenerPayload) => <any> options
);

export const restart:ActionCreator<types.RestartPayload> = createFSA<types.RestartPayload>(
    actionTypes.RESTART,
    (options:types.RestartPayload) => <any> options
);

export const scheduleRestart:ActionCreator<types.ScheduleRestartPayload> = createFSA<types.ScheduleRestartPayload>(
    actionTypes.APPLICATION_SCHEDULE_RESTART,
    (options:types.ScheduleRestartPayload) => <any> options
);


export const setShortcuts:ActionCreator<types.SetShortcutsPayload> = createFSA<types.SetShortcutsPayload>(
    actionTypes.APPLICATION_SET_SHORTCUTS,
    (options:types.SetShortcutsPayload) => <any> options
);

export const setTrayIcon:ActionCreator<types.SetTrayIconPayload> = createFSA<types.SetTrayIconPayload>(
    actionTypes.APPLICATION_SET_TRAY_ICON,
    (options:types.SetTrayIconPayload) => <any> options
);

export const setZoomLevel:ActionCreator<types.SetZoomLevelPayload> = createFSA<types.SetZoomLevelPayload>(
    actionTypes.APPLICATION_SET_ZOOM_LEVEL,
    (options:types.SetZoomLevelPayload) => <any> options
);

export const terminate:ActionCreator<types.TerminatePayload> = createFSA<types.TerminatePayload>(
    actionTypes.APPLICATION_TERMINATE,
    (options:types.TerminatePayload) => <any> options
);