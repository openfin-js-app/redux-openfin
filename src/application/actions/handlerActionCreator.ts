import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from "../../utils/createFSA";
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const startApplicationRes:ActionCreator<types.StartApplicationResPayload> = createFSA<types.StartApplicationResPayload>(
    actionTypes.START_APPLICATION_RES,
    (options:types.StartApplicationResPayload) => <any> options
);

export const getCurrentRes:ActionCreator<types.GetCurrentResPayload> = createFSA<types.GetCurrentResPayload>(
    actionTypes.APPLICATION_GET_CURRENT_RES,
    (options:types.GetCurrentResPayload) => <any> options
);

export const getWindowRes:ActionCreator<types.GetWindowResPayload> = createFSA<types.GetWindowResPayload>(
    actionTypes.APPLICATION_GET_WINDOW_RES,
    (options:types.GetWindowResPayload) => <any> options
);

export const wrapRes:ActionCreator<types.WrapResPayload> = createFSA<types.WrapResPayload>(
    actionTypes.APPLICATION_WRAP_RES,
    (options:types.WrapResPayload) => <any> options
);

export const addListenerRes:ActionCreator<types.AddListenerResPayload> = createFSA<types.AddListenerResPayload>(
    actionTypes.APPLICATION_ADD_LISTENER_RES,
    (options:types.AddListenerResPayload) => <any> options
);

export const quitRes:ActionCreator<types.QuitResPayload> = createFSA<types.QuitResPayload>(
    actionTypes.APPLICATION_QUIT_RES,
    (options:types.QuitResPayload) => <any> options
);

export const getChildWindowsRes:ActionCreator<types.GetChildWindowsResPayload> = createFSA<types.GetChildWindowsResPayload>(
    actionTypes.APPLICATION_GET_CHILD_WINDOWS_RES,
    (options:types.GetChildWindowsResPayload) => <any> options
);

export const getInfoRes:ActionCreator<types.GetInfoResPayload> = createFSA<types.GetInfoResPayload>(
    actionTypes.APPLICATION_GET_INFO_RES,
    (options:types.GetInfoResPayload) => <any> options
);

export const getShortcutsRes:ActionCreator<types.GetShortcutsResPayload> = createFSA<types.GetShortcutsResPayload>(
    actionTypes.APPLICATION_GET_SHORTCUTS_RES,
    (options:types.GetShortcutsResPayload) => <any> options
);

export const getTrayIconInfoRes:ActionCreator<types.GetTrayIconInfoResPayload> = createFSA<types.GetTrayIconInfoResPayload>(
    actionTypes.APPLICATION_GET_TRAY_ICON_INFO_RES,
    (options:types.GetTrayIconInfoResPayload) => <any> options
);

export const getZoomLevelRes:ActionCreator<types.GetZoomLevelResPayload> = createFSA<types.GetZoomLevelResPayload>(
    actionTypes.APPLICATION_GET_ZOOM_LEVEL_RES,
    (options:types.GetZoomLevelResPayload) => <any> options
);

export const isRunningRes:ActionCreator<types.IsRunningResPayload> = createFSA<types.IsRunningResPayload>(
    actionTypes.APPLICATION_IS_RUNNING_RES,
    (options:types.IsRunningResPayload) => <any> options
);

export const removeListenerRes:ActionCreator<types.RemoveListenerResPayload> = createFSA<types.RemoveListenerResPayload>(
    actionTypes.APPLICATION_REMOVE_LISTENER_RES,
    (options:types.RemoveListenerResPayload) => <any> options
);

export const restartRes:ActionCreator<types.RestartResPayload> = createFSA<types.RestartResPayload>(
    actionTypes.RESTART_RES,
    (options:types.RestartResPayload) => <any> options
);

export const scheduleRestartRes:ActionCreator<types.ScheduleRestartResPayload> = createFSA<types.ScheduleRestartResPayload>(
    actionTypes.APPLICATION_SCHEDULE_RESTART_RES,
    (options:types.ScheduleRestartResPayload) => <any> options
);


export const setShortcutsRes:ActionCreator<types.SetShortcutsResPayload> = createFSA<types.SetShortcutsResPayload>(
    actionTypes.APPLICATION_SET_SHORTCUTS_RES,
    (options:types.SetShortcutsResPayload) => <any> options
);

export const setTrayIconRes:ActionCreator<types.SetTrayIconResPayload> = createFSA<types.SetTrayIconResPayload>(
    actionTypes.APPLICATION_SET_TRAY_ICON_RES,
    (options:types.SetTrayIconResPayload) => <any> options
);

export const setZoomLevelRes:ActionCreator<types.SetZoomLevelResPayload> = createFSA<types.SetZoomLevelResPayload>(
    actionTypes.APPLICATION_SET_ZOOM_LEVEL_RES,
    (options:types.SetZoomLevelResPayload) => <any> options
);

export const terminateRes:ActionCreator<types.TerminateResPayload> = createFSA<types.TerminateResPayload>(
    actionTypes.APPLICATION_TERMINATE_RES,
    (options:types.TerminateResPayload) => <any> options
);
