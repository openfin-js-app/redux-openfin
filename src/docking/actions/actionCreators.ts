import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../DockingType";
import * as actionTypes from "./actionTypes";

export const undockAllWindows:ActionCreator<types.UndockAllWindowsPayload> = createFSA<types.UndockAllWindowsPayload>(
    actionTypes.UNDOCK_ALL_WINDOWS,
    (options:types.UndockAllWindowsPayload) => options
)