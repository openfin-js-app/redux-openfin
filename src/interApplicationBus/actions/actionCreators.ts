import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const publish:ActionCreator<types.PublishPayload> = createFSA<types.PublishPayload>(
        actionTypes.PUBLISH,
        (options:types.PublishPayload) => <any>{options}
);

export const subscribe:ActionCreator<types.SubscribePayload> = createFSA<types.SubscribePayload>(
        actionTypes.SUBSCRIBE,
        (options:types.SubscribePayload) => <any>{options}
);