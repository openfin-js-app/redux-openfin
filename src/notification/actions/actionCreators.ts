import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const createNotification:ActionCreator<Partial<types.CreateNotificationPayload>>
    =
createFSA<Partial<types.CreateNotificationPayload>>(
    actionTypes.CREATE_NOTIFICATION,
    (options:Partial<types.CreateNotificationPayload>) => <any>{options}
)