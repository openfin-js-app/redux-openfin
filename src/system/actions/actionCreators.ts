import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const getMachineId:ActionCreator<types.GetMachineIdPayload> = createFSA<types.GetMachineIdPayload>(
    actionTypes.GET_MACHINE_ID,
    (options:types.GetMachineIdPayload) => <any> options
);

export const getDeviceUserId:ActionCreator<types.GetDeviceUserIdPayload> = createFSA<types.GetDeviceUserIdPayload>(
    actionTypes.GET_DEVICE_USER_ID,
    (options:types.GetDeviceUserIdPayload) => <any> options
);

export const getMonitorInfo:ActionCreator<types.GetMonitorInfoPayload> = createFSA<types.GetMonitorInfoPayload>(
    actionTypes.GET_MONITOR_INFO,
    (options:types.GetMonitorInfoPayload) => <any> options
);

export const getVersion:ActionCreator<types.GetVersionPayload> = createFSA<types.GetVersionPayload>(
    actionTypes.GET_VERSION,
    (options:types.GetVersionPayload) => <any> options
);

export const getHostSpecs:ActionCreator<types.GetHostSpecsPayload> = createFSA<types.GetHostSpecsPayload>(
    actionTypes.GET_HOST_SPECS,
    (options:types.GetHostSpecsPayload) => <any> options
);


export const getEnvironmentVariable:ActionCreator<types.GetEnvironmentVariablePayload> = createFSA<types.GetEnvironmentVariablePayload>(
    actionTypes.GET_ENVIRONMENT_VARIABLE,
    (options:types.GetEnvironmentVariablePayload) => <any> options
);

export const clearCache:ActionCreator<types.ClearCachePayload> = createFSA<types.ClearCachePayload>(
    actionTypes.CLEAR_CACHE,
    (options:types.ClearCachePayload) => <any> options
);