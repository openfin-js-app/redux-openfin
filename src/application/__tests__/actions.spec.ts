import configureStore from 'redux-mock-store'
import * as Actions from '../actions/actionCreators';

const mockStore = configureStore();
const initialState = {};

const store = mockStore(initialState);

describe('Application pkg actions',()=>{

    describe('Request actions',()=>{

        it('close action',()=>{
            store.dispatch(Actions.close({force:true}));
            expect(store.getActions()).toMatchSnapshot();
        })

    });

});