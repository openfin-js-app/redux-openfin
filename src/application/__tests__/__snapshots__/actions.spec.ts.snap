// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application pkg actions Request actions close action 1`] = `
Array [
  Object {
    "error": undefined,
    "payload": Object {
      "force": true,
    },
    "type": "REDUX_OPENFIN_REQ::APPLICATION_CLOSE",
  },
]
`;
